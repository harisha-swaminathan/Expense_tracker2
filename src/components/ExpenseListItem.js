import React from "react";
import { Link } from "react-router-dom";
import { format } from "date-fns";
import numeral from "numeral";

const ExpenseListItem = ({description, amount, createdAt, id}) => {
   return  (
        <>
            <Link to={`/edit/${id}`}>
             <h3>{description}</h3>
            </Link> <br/>
            <p>
                {numeral(amount/100).format('$0,0.00')} 
                -
                {format(new Date(createdAt), "MMMM do, yyyy")}
            </p>
        </>
    );
}


export default ExpenseListItem;