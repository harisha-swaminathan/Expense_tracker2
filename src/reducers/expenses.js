const expenseReducerDefaultState = [];

const expensesReducer = (state = expenseReducerDefaultState, action) => {
    switch(action.type) {
        case 'ADD_EXPENSE':
            return [...state, action.expense];
        case 'DELETE_EXPENSE':
            return state.filter((expense) => expense.id !== action.id);
        case 'EDIT_EXPENSE':
            return state.map((expense) => {
                if(expense.id === action.id){
                    return {
                        ...expense,
                        ...action.updates
                    }
                } else {
                    return expense;
                }
            })
        default:
            return state;
    }
};

export default expensesReducer;