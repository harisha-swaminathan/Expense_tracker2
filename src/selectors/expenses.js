import { isBefore, isAfter } from "date-fns";


const getVisibleExpenses = (expenses, {text, sortBy, startDate, endDate}) => {
    return expenses.filter((expense) => {
      
        const description = expense.description.toLowerCase();
        const textMatch = typeof text !== 'string' || description.includes(text.toLowerCase());
        const startDateMatch = !startDate || isAfter(expense.createdAt, startDate); 
        const endDateMatch =  !endDate || isBefore(expense.createdAt, endDate);

        return textMatch && startDateMatch && endDateMatch;
    }).sort((expense1, expense2) => {
        if(sortBy == 'date'){
            return expense1.createdAt < expense2.createdAt ? 1 : -1;
        } else {
            return expense1.amount < expense2.amount ? 1 : -1;
        }
    })
};

export default getVisibleExpenses;