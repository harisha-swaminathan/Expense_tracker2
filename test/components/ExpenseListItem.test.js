import React from 'react';
import { render } from '@testing-library/react';
import ExpenseListItem from "../../src/components/ExpenseListItem";
import { BrowserRouter } from 'react-router-dom';
import expenses from '../fixtures/expenses';

test('expenseList is rendered with expenses as expected', () =>{
    const { asFragment } = render(<BrowserRouter> <ExpenseListItem key={expenses[0].id} {...expenses[0]}/> </BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
});
