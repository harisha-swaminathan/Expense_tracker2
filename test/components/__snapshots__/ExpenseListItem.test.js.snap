// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expenseList is rendered with expenses as expected 1`] = `
<DocumentFragment>
   
  <a
    href="/edit/1"
  >
    <h3>
      celine bag
    </h3>
  </a>
   
  <br />
  <p>
    $8.50-December 31st, 1969
  </p>
   
</DocumentFragment>
`;
